mysql-test/main/rpl_mysqldump_slave.test

```diff
@@ -83,6 +83,46 @@ DROP TABLE t2;
 --replace_regex /MASTER_LOG_POS=[0-9]+/MASTER_LOG_POS=BINLOG_START/
 --exec $MYSQL_DUMP_SLAVE --compact --master-data --single-transaction --gtid test
 
+#
+# MDEV-32611 Added test for mysqldump --delete-master-logs option.
+# This options is alias of 
+# get binlogs: show master status -> flush logs -> purge binary logs to <new_binlog>
+# sequence and this test is derived using the same pattern.
+#
+
+connection master;
+
+CREATE TABLE t (
+    id int
+);
+
+insert into t values (1);
+insert into t values (2);
+
+drop table t;
+
+--let $predump_binlog_filename= query_get_value(SHOW MASTER STATUS, File, 1)
+
+# Execute mysqldump with delete-master-logs option
+--replace_regex /MASTER_LOG_POS=[0-9]+/MASTER_LOG_POS=BINLOG_START/
+--exec $MYSQL_DUMP --compact --no-create-info --no-data --delete-master-logs test
+
+--let $postdump_binlog_filename= query_get_value(SHOW MASTER STATUS, File, 1)
+
+--let $postdump_first_binary_log_filename= query_get_value(SHOW BINARY LOGS, Log_name, 1)
```

Comment:
I think one of this is redundant (`show master status` or `show binary logs`).

Reply:
They check different aspects of the behavior. The check with `show master status` verifies `flush logs` occurred, and the check with `show binary logs` verifies `purge binary logs` occurred.